### DISCUSSION TITLE
Feature request: "corner controls" to not scale equally in both directions

### DISCUSSION THREAD

**Discussion by ptgamr** [fabricjs/fabric.js#7210] [Q&A] [answered]
At the moment, when I drag the control in the corners horizontally or vertically, the shape scale in both directions with an equal amount.

Would it be possible that we can add a flag to change this behavior so we can do things like this: (note, this is not implemented with Fabric.js)

![scaling](https://user-images.githubusercontent.com/1619740/125150596-185c2200-e195-11eb-9c39-80ac8288652b.gif)

Feel much more natural this way.

**Comment by melchiar** [comment, marked as answer]
I believe this is what you're looking for:

`canvas.uniformScaling = false;`

http://fabricjs.com/docs/fabric.Canvas.html#uniformScaling

**Reply by ptgamr** [reply]
Wow, thanks a lot @melchiar 

~~However, is there a way that I can make it the default behavior when resizing? (ie. doesn't have to press any key) At the moment I have to specify the `uniScaleKey` to have this feature.~~

Please ignore, I figure it out, just setting `uniformScaling`  to false do it for me. 

Thanks again!

**Comment by asturur** [comment]
Not entirely sure what are you doing, but please also consider that maybe you want to use the controls to change width and height if you are working just with rectangles to highlight some stuff.
